### PR TITLE
Cleanup sonar issues 6 - potential NPEs

### PR DESCRIPTION
**Pull request by jbee**
Major sonar issues (bugs). Focussed on potential NPEs. Sometimes I also cleaned other problems found at the same location. As before I annotate the changes that are less trivial and require bit of context.

### CODE REVIEW

File: appserver/ha/ha-file-store/src/main/java/org/glassfish/ha/store/adapter/file/FileBackingStore.java [hunk 1]
```diff
@@ -326,33 +326,16 @@ private boolean removeFile(final File file) {
...
-        ByteArrayOutputStream bos = new ByteArrayOutputStream();
-        ObjectOutputStream oos = null;
-        try {
-            oos = new ObjectOutputStream(bos);
+        try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(bos)) {
             oos.writeObject(value);
             oos.flush();
-            data = bos.toByteArray();
+            return bos.toByteArray();
```
Comment (jbee) [inline appserver/ha/ha-file-store/src/main/java/org/glassfish/ha/store/adapter/file/FileBackingStore.java]: static analysis now should be able to work out that result is never null. Also removed the key that wasn't used and that made this somewhat confusing. The caller for this is the like change above.

File: appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/DMManagedConnectionFactory.java [hunk 2]
```diff
@@ -191,34 +191,13 @@
      * @param key
      * @return
      */
-    private String getParsedKey(String key) throws ResourceException {
```
Comment (jbee) [inline appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/DMManagedConnectionFactory.java]: This could return `null` for any key not starting with `"set"` as well as the `null` key but the caller did not expect to get back null ever and fixing it there didn't make much sense. Also this method already had a test checking for null or empty key in case it was removing the "set" prefix so it seem logical that passing `null` or the empty string to this should equally cause the `ResourceException`.

File: appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionFactoryImpl.java [hunk 3]
```diff
@@ -600,33 +600,35 @@ private void detectSqlTraceListeners() {
...
-                            try {
-
-                                constructors = listenerClass.getConstructors();
-                                for (Constructor constructor : constructors) {
-                                    parameterTypes = constructor.getParameterTypes();
-                                    //For now only the no argument constructors are allowed.
-                                    //TODO should this be documented?
-                                    if (parameterTypes != null && parameterTypes.length == 0) {
-                                        listener = (SQLTraceListener) constructor.newInstance(initargs);
+                    if (listenerClass != null) {
```
Comment (jbee) [inline appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionFactoryImpl.java]: `listenerClass` can be null which now will not try to register any listener (which would cause a NPE)

File: appserver/payara-appserver-modules/payara-micro-service/src/main/java/fish/payara/appserver/micro/services/data/InstanceDescriptorImpl.java [hunk 4]
```diff
@@ -176,18 +176,10 @@ public void removeApplication(ApplicationDescriptor applicationInfo) {
...
     /**
      * Overrides equals purely based on the UUID value
-     *
-     * @param obj
-     * @return
      */
     @Override
     public boolean equals(Object obj) {
-        boolean result = false;
-        if (InstanceDescriptorImpl.class.isInstance(obj)) {
```
Comment (jbee) [inline appserver/payara-appserver-modules/payara-micro-service/src/main/java/fish/payara/appserver/micro/services/data/InstanceDescriptorImpl.java]: sonar didn't like use of `isInstance` over `instanceof`. I agree :)

File: appserver/persistence/cmp/ejb-mapping/src/main/java/com/sun/jdo/api/persistence/mapping/ejb/SunCmpMappingsUtils.java [hunk 5]
```diff
@@ -176,24 +176,23 @@ else if (l.size() > 1)
...
-        if (len > 0)
-            retVal = new ArrayList(); 
-        for (int i = 0; i < len; i++) {
+        if (props == null) {
+            return null;
+        }
+        List<BaseBean> retVal = new ArrayList<>(); 
+        for (int i = 0; i < props.length; i++) {
             // get the bean that is the property's parent.
-            BaseBean candidate = gm.getPropertyParent(props[i]);
```
Comment (jbee) [inline appserver/persistence/cmp/ejb-mapping/src/main/java/com/sun/jdo/api/persistence/mapping/ejb/SunCmpMappingsUtils.java]: sonar can't do index based exclusion so I rewrote the body to both make it more readable for humans and tools.

File: appserver/persistence/cmp/enhancer/src/main/java/com/sun/jdo/api/persistence/enhancer/PersistenceLauncher.java [hunk 6]
```diff
@@ -192,8 +192,10 @@ public static void main(String[] args)
             out.println("    }");//NOI18N
             out.println("    targetClassname = " + targetClassname);//NOI18N
             out.print("    targetClassArgs = { ");//NOI18N
-            for (int i = 0; i < targetClassArgs.length; i++) {
-                out.print(targetClassArgs[i] + " ");//NOI18N
```
Comment (jbee) [inline appserver/persistence/cmp/enhancer/src/main/java/com/sun/jdo/api/persistence/enhancer/PersistenceLauncher.java]: Again index based exclusion needs some help so I added a null check.

**Comment by jbee** [comment]
jenkins test please

**Comment by jbee** [comment]
jenkins test please

File: appserver/ha/ha-file-store/src/main/java/org/glassfish/ha/store/adapter/file/FileBackingStore.java [hunk 7]
```diff
@@ -427,17 +388,6 @@ private void writetoFile(K sessionKey, String fileName, byte[] data)
             }
             String errMsg = "Could not save session: " + sessionKey;
             throw new BackingStoreException(errMsg, ex);
-        } finally {
```
Comment (MattGill98) [inline appserver/ha/ha-file-store/src/main/java/org/glassfish/ha/store/adapter/file/FileBackingStore.java]: Should we keep the failure logging here with a catch?

Comment (jbee) [reply appserver/ha/ha-file-store/src/main/java/org/glassfish/ha/store/adapter/file/FileBackingStore.java]: I wouldn't say so. Its logging of exceptions occurring in something that is backed into try-with-resource. I would think that these were only added because the author didn't want to just ignore a cached exception.

Comment (jbee) [reply appserver/ha/ha-file-store/src/main/java/org/glassfish/ha/store/adapter/file/FileBackingStore.java]: Maybe a further clarification: Using try-with-resources is a change of behaviour since exceptions occurring while closing will no longer be caught. My assumptions here was that the author was using the replaced code as the pattern to use for IO and I replaced it with the new pattern, the try-with-resouce.
